server/storage/backend/batch_tx.go

```diff
@@ -269,7 +280,9 @@ func (t *batchTx) commit(stop bool) {
 		start := time.Now()
 
 		// gofail: var beforeCommit struct{}
+		cancel := watchdog.Register("batchTx commit")
```

Comment:
No sure that plan about how to test it. Maybe the watchdog.Register should be called before beforeCommit here, so that the test can inject the `sleep` failpoint? Or introduces the underlying fs/block to simulate the real stall, like https://github.com/etcd-io/etcd/pull/14691/files

Reply:
Makes sense & updated. I also need to add more failpoint(s) later.